swift/apple/FirezoneKit/Sources/FirezoneKit/AuthClient/AuthClient.swift

```diff
@@ -95,6 +121,34 @@ private final class WebAuthenticationSession: NSObject,
     }
   }
 
+  static func createRandomHexString(byteCount: Int) throws -> String {
+    var bytes = [Int8](repeating: 0, count: byteCount)
+    let status = SecRandomCopyBytes(kSecRandomDefault, bytes.count, &bytes)
+
+    guard status == errSecSuccess else {
+      throw AuthClientError.randomNumberGenerationFailure(errorStatus: status)
+    }
+
+    return bytes.map { String(format: "%02hhx", $0) }.joined()
+  }
+
+  static func areStringsEqualConstantTime(_ string1: String, _ string2: String) -> Bool {
```

Comment:
It would be hard to test the constant time part of this, but it would be good to test here:

- `"" == ""`
- `"foo" == "foo"`
- `"foo" != "bar"`